src/modules/Auth/features/Login/MFAForm/RecoveryCodeForm.tsx

```diff
@@ -0,0 +1,168 @@
+import { yupResolver } from '@hookform/resolvers/yup';
+import { ChangeEvent, useEffect, useRef } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import { TextField } from '@mui/material';
+
+import { recoveryCodeSchema } from './MFAForm.schema';
+import {
+  StyledMFAContainer,
+  StyledMFAForm,
+  StyledMFAHeadline,
+  StyledMFASubheader,
+  StyledMFAController,
+  StyledMFAButton,
+  StyledBackButton,
+} from './MFAForm.styles';
+import { useMFAVerification } from './useMFAVerification';
+
+interface RecoveryCodeFormData {
+  code: string;
+}
+
+interface RecoveryCodeFormProps {
+  onSwitchToTOTP?: () => void;
+  onBackToLogin?: () => void;
+}
+
+export const RecoveryCodeForm = ({ onSwitchToTOTP, onBackToLogin }: RecoveryCodeFormProps) => {
+  const { t } = useTranslation('app');
+
+  const { error, displayError, isSessionExpired, isSubmitting, verifyCode, clearError, cleanup } =
+    useMFAVerification('recovery');
+  const isUserTypingRef = useRef(false);
+
+  const {
+    handleSubmit,
+    control,
+    setValue,
+    watch,
+    formState: { errors },
+  } = useForm<RecoveryCodeFormData>({
+    resolver: yupResolver(recoveryCodeSchema()),
+    defaultValues: {
+      code: '',
+    },
+  });
+
+  const code = watch('code');
+
+  // Clear error only when user is actually typing
+  useEffect(() => {
+    if (error && code.length > 0 && isUserTypingRef.current) {
+      clearError();
+    }
+  }, [code, error, clearError]);
+
+  // Cleanup on unmount
+  useEffect(() => cleanup, [cleanup]);
+
+  const onSubmit = async (data: RecoveryCodeFormData) => {
+    if (isSessionExpired) return;
+    isUserTypingRef.current = false; // Mark that we're not typing
+    const success = await verifyCode(data.code);
+    if (!success) {
+      // Clear the input on error so user can try again
+      setValue('code', '');
+    }
+  };
+
+  const handleBackToTOTP = () => {
+    if (onSwitchToTOTP) {
+      onSwitchToTOTP();
+    }
+  };
+
+  // Auto-format recovery code input
+  const formatRecoveryCode = (value: string) => {
```

Comment:
Could we remove `formatRecoveryCode` here and import from **mfa.utils.ts**? Keeping the function in **mfa.utils.ts** seems like the right approach so we can reuse it elsewhere.